
<launch>
    <!-- pololu servo controller node -->
    <node pkg="ros_pololu_servo" type="ros_pololu_servo_node" name="ros_pololu_servo_node" output="screen">
        <param name="pololu_motors_yaml" value="$(find ros_pololu_servo)/launch/pololu_motors.yaml" />
	      <param name="pololu_sensors_yaml" value="$(find ros_pololu_servo)/launch/pololu_sensors.yaml" />
        <param name="port_name" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
        <param name="rate_hz" value="10" />
        <param name="daisy_chain" value="false" />
    </node>

    <!-- steering pid -->
    <node name="steering" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.1" />
        <param name="upper_limit" value="10" />
        <param name="lower_limit" value="-10" />
        <param name="windup_limit" value="10" />
        <param name="cutoff_frequency" value="20" />
        <param name="max_loop_frequency" value="105.0" />
        <param name="min_loop_frequency" value="95.0" />
        <param name="topic_from_plant" value="steering_plant" />
        <param name="setpoint_topic" value="steering_setpoint" />
        <param name="topic_from_controller" value="steering_effort" />
        <param name="pid_enable_topic" value="steering_pid_enable" />
    </node>

    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
      args="/steering_effort/data /steering_plant/data /steering_setpoint/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <!-- motor pid -->
    <node name="steering" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.1" />
        <param name="upper_limit" value="10" />
        <param name="lower_limit" value="-10" />
        <param name="windup_limit" value="10" />
        <param name="cutoff_frequency" value="20" />
        <param name="max_loop_frequency" value="105.0" />
        <param name="min_loop_frequency" value="95.0" />
        <param name="topic_from_plant" value="motor_plant" />
        <param name="setpoint_topic" value="motor_setpoint" />
        <param name="topic_from_controller" value="motor_effort" />
        <param name="pid_enable_topic" value="motor_pid_enable" />
    </node>

    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
      args="/motor_effort/data /motor_plant/data /motor_setpoint/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <!-- robot controller node -->
    <node pkg="oto_control" type="oto_control_node" name="oto_control_node" output="screen">
    </node>
</launch>
